include/random_walks/exponential_hamiltonian_monte_carlo_walk.hpp

```diff
@@ -0,0 +1,205 @@
+                      unsigned int const& walk_length,
+                      RandomNumberGenerator &rng)
+    {
+        unsigned int n = P.dimension();
+        NT T;
+        int failures = 0, it;
+        Point p0;
+
+        for (auto j=0u; j<walk_length; ++j)
+        {
+            do {
+                failures++;
+                if (failures == 1000) {
+                    return false;
+                }
+                T = -std::log(rng.sample_urdist()) * _Len;
+                _v = GetDirection<Point>::apply(n, rng, false);
+                p0 = _p;
+                it = 0;
+                while (it < 200*n)
+                {
+                    auto pbpair = P.quadratic_positive_intersect(_p, _v, _Ac, _Temp, _lambdas,
+                                                             _Av, _lambda_prev, _facet_prev);
+                    if (T <= pbpair.first || pbpair.second < 0) {
+                        _p += ((T * T) / (-2.0*_Temp)) *_c + (T * _v);
+                        _lambda_prev = T;
+                        break;
+                    }
+                    _lambda_prev = pbpair.first;
+                    _p += ((_lambda_prev * _lambda_prev) / (-2.0*_Temp)) *_c + (_lambda_prev * _v);
+                    T -= _lambda_prev;
+                    _v += (-_lambda_prev/_Temp) * _c;
+                    P.compute_reflection(_v, _p, pbpair.second);
+                    it++;
+                }
+                
+            } while (P.is_in(_p, _tol) == 0);
+            if (it == 200*n){
+                _p = p0;
+            }
+        }
+        p = _p;
+        return true;
+    }
+
+    inline void update_delta(NT L)
+    {
+        _Len = L;
+    }
+
+private :
+
+    template
+    <
+        typename GenericPolytope
+    >
+    inline void initialize(GenericPolytope const& P,
+                           Point const& p,
+                           RandomNumberGenerator &rng)
+    {
+        unsigned int n = P.dimension();
+        NT T;
+        _lambdas.setZero(P.num_of_hyperplanes());
+        _Av.setZero(P.num_of_hyperplanes());
+        _p = p;
+        _v = GetDirection<Point>::apply(n, rng, false);
+        
+        do {
+            T = -std::log(rng.sample_urdist()) * _Len;
+            Point p0 = _p;
+            int it = 0;
+
+            std::pair<NT, int> pbpair
+                    = P.quadratic_positive_intersect(_p, _v, _Ac, _Temp, _lambdas, _Av, _facet_prev);
+            if (T <= pbpair.first || pbpair.second < 0) {
+                _p += ((T * T) / (-2.0*_Temp)) *_c + (T * _v);
+                _lambda_prev = T;
+                return;
+            }
+            _lambda_prev = pbpair.first;
+            _p += ((_lambda_prev * _lambda_prev) / (-2.0*_Temp)) *_c + (_lambda_prev * _v);
+            _v += (-_lambda_prev/_Temp) * _c;
+            T -= _lambda_prev;
+            P.compute_reflection(_v, _p, pbpair.second);
+
+            while (it <= 100*n)
```

Comment:
why `100*n`

Reply:
It is the same with `AcceleratedBilliardWalk`.